File: CS/15_Document/SetMagnificationToFitHeight.cs
using Spire.Pdf;
using Spire.Pdf.Actions;
using Spire.Pdf.General;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;

namespace SetMagnificationToFitHeight
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void button1_Click(object sender, EventArgs e)
        {
            // Create a pdf document and load file from disk
            PdfDocument myPdf = new PdfDocument();
            myPdf.LoadFromFile(@"..\..\..\..\..\..\Data\SampleB_1.pdf");

            // Get the first page
            PdfPageBase page = myPdf.Pages[0];

            // Create a PdfDestination with specific page, location
            PdfDestination dest = new PdfDestination(page, new PointF(-40f, -40f));

            // Set the Magnification to Fit-Height
            dest.Mode = PdfDestinationMode.FitV;

            //Create GoToAction with dest
            PdfGoToAction gotoaction = new PdfGoToAction(dest);

            // Set open action
            myPdf.AfterOpenAction = gotoaction;
            myPdf.ViewerPreferences.PageMode = PdfPageMode.UseOutlines;

            // Save the file
            myPdf.SaveToFile("FitHeight.pdf");
            myPdf.Close();

            // Launch the pdf file
            PDFDocumentViewer("FitHeight.pdf");
        }

        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
        
    }
}

File: VB/15_Document/SetMagnificationToFitHeight.vb
Imports Spire.Pdf
Imports Spire.Pdf.Actions
Imports Spire.Pdf.General

Namespace SetMagnificationToFitHeight
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub
        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new PdfDocument object
            Dim myPdf As New PdfDocument()

            ' Load an existing PDF document from file
            myPdf.LoadFromFile("..\..\..\..\..\..\Data\SampleB_1.pdf")

            ' Get the first page of the loaded PDF document
            Dim page As PdfPageBase = myPdf.Pages(0)

            ' Create a new destination for the page with a specific location
            Dim dest As New PdfDestination(page, New PointF(-40.0F, -40.0F))

            ' Set the mode of the destination to fit the page vertically
            dest.Mode = PdfDestinationMode.FitV

            ' Create a new GoTo action with the destination
            Dim gotoaction As New PdfGoToAction(dest)

            ' Set the GoTo action as the after open action for the document
            myPdf.AfterOpenAction = gotoaction

            ' Set the page mode of the viewer preferences to use outlines
            myPdf.ViewerPreferences.PageMode = PdfPageMode.UseOutlines

            ' Save the modified document to a new PDF file
            myPdf.SaveToFile("FitHeight.pdf")

            ' Close the document
            myPdf.Close()

            ' Launch the pdf file
            PDFDocumentViewer("FitHeight.pdf")
        End Sub

        Private Sub PDFDocumentViewer(ByVal fileName As String)
            Try
                Process.Start(fileName)
            Catch
            End Try
        End Sub

    End Class
End Namespace
